pkg/gadgets/trace/exec/tracer/bpf/execsnoop.bpf.c

```diff
@@ -89,23 +89,24 @@ int ig_execve_e(struct trace_event_raw_sys_enter* ctx)
 	event->args_count++;
 	#pragma unroll
 	for (i = 1; i < TOTAL_MAX_ARGS && i < max_args; i++) {
-		bpf_probe_read_user(&argp, sizeof(argp), &args[i]);
-		if (!argp)
+		ret = bpf_probe_read_user(&argp, sizeof(argp), &args[i]);
+		if (ret != 0 || !argp)
 			return 0;
 
 		if (event->args_size > LAST_ARG)
 			return 0;
 
 		ret = bpf_probe_read_user_str(&event->args[event->args_size], ARGSIZE, argp);
-		if (ret > ARGSIZE)
+		if (ret > 0 && ret <= ARGSIZE) {
+			event->args_count++;
+			event->args_size += ret;
+		} else {
 			return 0;
-
-		event->args_count++;
-		event->args_size += ret;
+		}
```

Comment:
It should be `ret < 0 || ret >= ARGSIZE` as it never returns > ARGSIZE -> https://github.com/inspektor-gadget/inspektor-gadget/pull/1709#discussion_r1232181778

Reply:
I think it should just be `ret < 0`. If `ret == ARGSIZE`, this is not an error, we were just lucky that the argument fits perfectly in the buffer.

Reply:
mmm, documentation says `If the string length is larger than size, just size-1 bytes are copied and the last byte is set to NUL.`, so in that case the string is trimmed, right? So I think we can't distinguish the case we're we're lucky (the string fits perfectly in the buffer) and the case the string is trimmed.

Reply:
We can't distinguish those two cases, but I think we don't want to: if the string is cut, we still want to return the partial string instead of returning nothing.

Reply:
I agree ^^^.